easybuild/easyblocks/generic/configuremake.py

```diff
@@ -83,13 +87,30 @@ def configure_step(self, cmd_prefix=''):
         if prefix_opt is None:
             prefix_opt = '--prefix='
 
-        cmd = "%(preconfigopts)s %(cmd_prefix)s./configure %(prefix_opt)s%(installdir)s %(configopts)s" % {
-            'preconfigopts': self.cfg['preconfigopts'],
-            'cmd_prefix': cmd_prefix,
-            'prefix_opt': prefix_opt,
-            'installdir': self.installdir,
-            'configopts': self.cfg['configopts'],
-        }
+        configure_command = cmd_prefix + './configure'
+
+        # Avoid using config.guess from an Autoconf generated package as it is frequently out of date, use the version
+        # shipped with EB instead and provide the result to the configure command
+        build_type_option = ''
+        # possible that the configure_command is generated using preconfigopts...we're at the mercy of the gods then
+        if os.path.isfile(configure_command):
+            if 'Generated by GNU Autoconf' in open(configure_command).read():
+                build_type = self.cfg.get('build_type')
+                if build_type is None:
+                    config_guess_path, _ = run_cmd('which config.guess')
+                    config_guess_path = config_guess_path.strip()
```

Comment:
there's a dedicated `which` function in `filetools`, please use that (on some stripped down systems `which` is not installed)

```python
config_guess_path = which('config.guess')
```

(no need to `.strip()` either then)